src/unity/lib/image_util.cpp

```diff
@@ -230,13 +235,41 @@ std::vector<std::string> get_directory_files(std::string url, bool recursive) {
   return ret;
 }
 
+
+bool has_ending (std::string fullString, std::string ending) {
+    if (fullString.length() >= ending.length()) {
+        return (0 == fullString.compare(fullString.length() - ending.length(), ending.length(), ending));
+    } else {
+        return false;
+    }
+}
+
 /**
  * Construct an sframe of flex_images, with url pointing to directory where images reside. 
  */
 std::shared_ptr<unity_sframe> load_images(std::string url, std::string format, bool with_path, bool recursive,
                                           bool ignore_failure, bool random_order) {
     log_func_entry();
-    std::vector<std::string> all_files = get_directory_files(url, recursive);
+
+    std::vector<std::string> all_files;
+    if (has_ending(url, std::string(".jpg")) 
+      || has_ending(url, std::string(".png"))) {
```

Comment:
we also support ".jpeg".

Reply:
And the case-sensitive variants of them. You may want to make `has_ending` case insensitive always (though the name should probably change at that point), or make these comparisons case insensitive (if you think `has_ending` needs to be case sensitive generally).